test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -97,14 +119,14 @@ TEST_F(XdsOverrideHostTest, SwapChildPolicy) {
             absl::OkStatus());
   for (absl::string_view address : kAddresses) {
     auto subchannel = FindSubchannel({address});
-    ASSERT_TRUE(subchannel);
+    ASSERT_NE(subchannel, nullptr);
     ASSERT_TRUE(subchannel->ConnectionRequested());
```

Comment:
This should not be passing.  The first subchannel should already have been reporting READY before round_robin starts up, so it should never request a connection on the first subchannel, only on the second one.

If this is actually passing as currently written, then I think there's a bug somewhere.

Reply:
For this PR, let's just change the code not to check this, since it shouldn't be happening.  The first subchannel should not be asked to connect, and we should not be having it report CONNECTING again, because it was already in READY state.

I think that after we switch from pick_first to round_robin, we should initially get a new picker that repeatedly returns only the first subchannel, because that subchannel was already connected from when we were using pick_first.  Then we can check that round_robin requests a connection on the second subchannel, then have that subchannel report CONNECTING followed by READY, and then we can expect a picker that round-robins between the two.

In other words, I think what we want here is this:

```
  // 2. Now we switch to a round-robin
  EXPECT_EQ(ApplyUpdate(BuildUpdate(kAddresses,
                                    MakeXdsOverrideHostConfig("round_robin")),
                        policy_.get()),
            absl::OkStatus());
  // First subchannel should already be connected, so round_robin
  // should initially return a picker that repeatedly returns the same subchannel.
  picker = ExpectState(GRPC_CHANNEL_READY);
  ASSERT_NE(picker, nullptr);
  EXPECT_EQ(kAddresses[0], ExpectPickComplete(picker.get());
  EXPECT_EQ(kAddresses[0], ExpectPickComplete(picker.get());
  // The round_robin policy should have requested a connection on the second subchannel.
  auto* subchannel = FindSubchannel({kAddresses[1]});
  ASSERT_NE(subchannel, nullptr);
  EXPECT_TRUE(subchannel->ConnectionRequested());
  // The subchannel reports CONNECTING.  We'll get a new READY picker, but it will
  // continue to hand out only the first subchannel.
  subchannel->SetConnectivityState(GRPC_CHANNEL_CONNECTING);
  picker = ExpectState(GRPC_CHANNEL_READY);
  ASSERT_NE(picker, nullptr);
  EXPECT_EQ(kAddresses[0], ExpectPickComplete(picker.get());
  EXPECT_EQ(kAddresses[0], ExpectPickComplete(picker.get());  
  // Now the subchannel reports READY.  We get a new picker that returns both subchannels.
  subchannel->SetConnectivityState(GRPC_CHANNEL_READY);
  picker = ExpectState(GRPC_CHANNEL_READY);
  ASSERT_NE(picker, nullptr);
  std::set<std::string> picked;
  for (size_t i = 0; i < kAddresses.size(); i++) {
    auto pick = ExpectPickComplete(picker.get());
    ASSERT_TRUE(pick.has_value());
    picked.insert(*pick);
  }
  EXPECT_THAT(picked, ::testing::UnorderedElementsAreArray(kAddresses));
```